### PR TITLE
Implement Wireframe Design and Model Methods

### PR DESCRIPTION
**Pull request by MarcoDDM**
**Description:**
This pull request focuses on implementing the wireframe design and enhancing model methods for the Blog App. The goal is to ensure proper functionality and navigation based on the wireframes provided. Additionally, model methods have been enhanced to process data effectively.

**Changes Made:**
- Implemented wireframe design for various pages: All Users, Single User, All Posts by a given User, and Single Post.
- Updated styling to match the design, with a primary focus on functionality.
- Created model methods to handle data processing:
  - Improved the `recent_posts` method to retrieve the most recent posts.
  - Added methods to calculate the counts of posts, comments, and likes for each user.
  - Utilized subqueries to calculate the number of comments and likes associated with posts.
- Reviewed and optimized database queries to ensure efficient data retrieval.
- Added sample data to the database to facilitate testing and visualization of posts and comments.

**Screenshots**
 Added screenshots for the code reviewer convenience.
- Users:
views_users
- User:
views_users2
- Posts:
views_posts
- Post:
views_post

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,17 @@
-<h1>List of users</h1>
+<main>
+  <section>
+    <ul>
+      <% @users.each do |user| %>
+        <%= link_to user_user_posts_path(user) do %>
```
Comment (Lucas-Masaba) [inline app/views/users/index.html.erb]: - [x] Clicking on a user should take a user to the "users show page"(/users/1) but I am instead taken to the "posts index page"(/users/1/user_posts).

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,14 @@
...
+    <ul>
+      <% @user.recent_posts.each do |post| %>
+        <%= link_to user_user_post_path(@user, post) do %>
+          <li class="card flexColumn hover">
+            <h3 class="left"><%= post.title %></h3>
+            <h4 class="left"><%= post.text %></h4>
+          </li>
+        <% end %>
+      <% end %>
+    </ul>
```
Comment (Lucas-Masaba) [inline app/views/users/show.html.erb:31]: - [x] There should be a button that takes a user to the "posts index page"(/users/1/posts) called "See All Posts" as seen in the provided wireframe
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @MarcoDDM  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Descriptive pull request message :scroll: 
- Well written commit messages :writing_hand: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @MarcoDDM         👍 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Rails features  ✔️
- [x] Kudus on fixing all changes suggested by the last reviewer   
- [x] Kudus on meeting the requirements   👏 


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**